modules/cudacodec/test/test_video.cpp

```diff
@@ -297,8 +293,15 @@ CUDA_TEST_P(Video, Reader)
         {cudacodec::ColorFormat::NV_NV12,1}
     };
 
-    std::string inputFile = std::string(cvtest::TS::ptr()->get_data_path()) + "../" + GET_PARAM(1);
-    cv::Ptr<cv::cudacodec::VideoReader> reader = cv::cudacodec::createVideoReader(inputFile);
+    std::string inputFile = std::string(cvtest::TS::ptr()->get_data_path()) + "../" + relativeFilePath;
+    cv::Ptr<cv::cudacodec::VideoReader> reader;
+    try {
+        reader = cv::cudacodec::createVideoReader(inputFile);
+    }
+    catch (const Exception& e) {
+        ASSERT_TRUE(relativeFilePath == "highgui/video/sample_322x242_15frames.yuv420p.libaom-av1.mp4");
```

Comment:
I propose:
```
if (relativeFilePath == "highgui/video/sample_322x242_15frames.yuv420p.libaom-av1.mp4")
    throw SkipTestException("AV1 decoding not supported in this version of FFmpeg - SKIP");
else
    throw e;
```
It's more informative for debugging in case of real exception.

Reply:
I should have paid more attention to the error

> [ERROR:1@1.660] global video_decoder.cpp:127 create Video source is not supported by hardware video decoder.

FFmpeg is failing to parse the file but `VideoReader` is falling back to `CuvidVideoSource` which is able to parse the file.  Then its failing because your GPU doesn't support AV1 decoding meaning the new error message in the test is wrong.

Because the exceptions from the decoder are caught and logged it is difficult without modification to assess what the exact cause of the failure is, hence the additional logging of the above message and the general exception which is thrown from the main thread

> error: (-2:Unspecified error) Parsing/Decoding video source failed, check GPU memory is available and GPU supports hardware decoding.

Should I at the very least modify the skip test exception to output the above instead of 

> "AV1 decoding not supported in this version of FFmpeg - SKIP"

?

Going forward I could modify the decoder to internally log that the codec (or other parameters) are not supported by the Nvidia hardware decoder making this information available at test time, what do you think?

Reply:
I should have added (seeing https://github.com/opencv/opencv/pull/23540#issuecomment-1598810357) that the error is also thrown after construction.  This will be rectified so that the exception is thrown in by the constructor once https://github.com/opencv/opencv_contrib/pull/3486 (itself a direct result of similar discussions like this we've had in the past) is merged.